python/packages/kagent-crewai/src/kagent/crewai/_memory.py

```diff
@@ -0,0 +1,110 @@
+import logging
+from typing import Any, Dict, List
+
+import httpx
+from pydantic import BaseModel, Field
+
+
+class KagentMemoryPayload(BaseModel):
+    session_id: str = Field(..., alias="thread_id")
+    user_id: str
+    memory_data: Dict[str, Any]
+
+
+class KagentMemoryResponse(BaseModel):
+    data: List[KagentMemoryPayload]
+
+
+class KagentMemoryStorage:
+    """
+    KagentMemoryStorage is a custom storage class for CrewAI's LongTermMemory.
+    It persists memory items to the Kagent backend, scoped by session_id and user_id.
+    """
+
+    def __init__(self, session_id: str, user_id: str, base_url: str = "http://localhost:8080"):
+        self.session_id = session_id
+        self.user_id = user_id
+        self.base_url = base_url
+
+    def save(self, task_description: str, metadata: dict, timestamp: str, score: float) -> None:
+        """
+        Saves a memory item to the Kagent backend.
+        The agent_id is expected to be in the metadata.
+        """
+        url = f"{self.base_url}/api/crewai/memory"
+        payload = KagentMemoryPayload(
+            session_id=self.session_id,
```

Comment:
My IDE is complaining that this should be `thread_id` as per pydantic, not sure if yours is also mad.